backend/pgnService/game/update.ts

```diff
@@ -144,60 +129,81 @@ function getRequest(event: APIGatewayProxyEventV2): UpdateGameRequest {
  * @param request The UpdateGameRequest to process.
  * @returns A GameUpdate based on the given request.
  */
-async function getGameUpdate(
-    request: UpdateGameRequest,
-): Promise<[GameUpdate | null, GameImportHeaders | null]> {
+async function getGameUpdate(request: UpdateGameRequest): Promise<GameUpdate> {
     const update: GameUpdate = {
         updatedAt: new Date().toISOString(),
     };
 
-    if (request.orientation) {
-        update.orientation = request.orientation;
-    }
     if (request.unlisted !== undefined) {
         update.unlisted = request.unlisted;
-        update.publishedAt = request.unlisted ? null : new Date().toISOString();
-        update.timelineId = request.unlisted
-            ? ''
-            : `${update.publishedAt?.split('T')[0]}_${uuidv4()}`;
+
+        if (request.unlisted) {
+            update.publishedAt = null;
+            update.timelineId = '';
+        } else {
+            update.publishedAt = new Date().toISOString();
+            update.timelineId = `${update.publishedAt?.split('T')[0]}_${uuidv4()}`;
+        }
+    }
+
+    if (request.orientation) {
+        update.orientation = request.orientation;
     }
 
+    let result: string | undefined = request.headers?.result;
     if (request.type) {
         const pgnText = (await getPgnTexts(request))[0];
         const game = getGame(undefined, pgnText, request.headers);
-        const headers = getImportHeaders(game);
-        if (!request.unlisted && headers) {
-            return [null, headers];
-        }
 
         update.white = game.white;
         update.black = game.black;
         update.date = game.date;
-        update.headers = game.headers;
         update.pgn = game.pgn;
+        update.headers = game.headers;
+
+        result = game.headers['Result'] ?? result;
+
+        if (isMissingData({ ...update, result }) && !update.unlisted) {
+            throw new ApiError({
+                statusCode: 400,
+                publicMessage: 'Published games can not be missing data',
+                privateMessage: 'update requested, but game was missing data',
+            });
+        }
     }
 
-    return [update, null];
+    return update;
 }
 
 /**
- * Returns the import headers for the Game, if it is missing data. Otherwise returns null.
- * @param game The game to get the import headers for.
- * @returns The import headers, if the game is missing data. Null otherwise.
+ * Strip name header
```

Comment:
Can this comment be a bit more descriptive on what it means to strip the name header?